postprocessing/comstockpostproc/comstock_query_builder.py

```diff
@@ -0,0 +1,328 @@
+
+        This matches the pattern from working buildstock queries that join timeseries data
+        with a weight view table for proper weighting and aggregation.
+
+        Args:
+            upgrade_id: Upgrade ID to filter on
+            enduses: List of end use columns to select
+            weight_view_table: Name of the weight view table (required - e.g., 'rtuadv_v11_md_agg_national_by_state_vu' or 'rtuadv_v11_md_agg_national_by_county_vu')
+            group_by: Additional columns to group by (e.g., ['build_existing_model.building_type'])
+            restrictions: List of (column, values) tuples for filtering
+            timestamp_grouping: How to group timestamps ('hour', 'day', 'month')
+            building_ids: Specific building IDs to filter on (optional)
+            include_sample_stats: Whether to include sample_count, units_count, rows_per_sample
+            include_area_normalized_cols: Whether to include weighted area and kwh_weighted columns for AMI comparison
+
+        Returns:
+            SQL query string
+        """
+
+        # Weight view table is required - cannot auto-assign without knowing state vs county aggregation
+        if weight_view_table is None:
+            raise ValueError("weight_view_table parameter is required. Cannot auto-assign without knowing geographic aggregation level (state vs county). "
+                           "Please provide the full weight view table name (e.g., 'your_dataset_md_agg_national_by_state_vu' or 'your_dataset_md_agg_national_by_county_vu')")
+
+        # Build timestamp grouping with date_trunc and time adjustment
+        if timestamp_grouping == 'hour':
+            time_select = f"date_trunc('hour', date_add('second', -900, {self.timeseries_table}.time)) AS time"
+            time_group = '1'
+        elif timestamp_grouping == 'day':
+            time_select = f"date_trunc('day', {self.timeseries_table}.time) AS time"
+            time_group = '1'
+        elif timestamp_grouping == 'month':
+            time_select = f"date_trunc('month', {self.timeseries_table}.time) AS time"
+            time_group = '1'
+        else:
+            time_select = f"{self.timeseries_table}.time AS time"
+            time_group = '1'
+
+        # Build SELECT clause with sample statistics
+        select_clauses = [time_select]
+
+        # Add group_by columns to SELECT (excluding 'time' which is already handled)
+        if group_by:
+            for col in group_by:
+                if col != 'time':
+                    select_clauses.append(f'{weight_view_table}."{col}"')
+
+        if include_sample_stats:
+            select_clauses.extend([
+                f"count(distinct({self.timeseries_table}.building_id)) AS sample_count",
+                f"(count(distinct({self.timeseries_table}.building_id)) * sum(1 * {weight_view_table}.weight)) / sum(1) AS units_count",
+                f"sum(1) / count(distinct({self.timeseries_table}.building_id)) AS rows_per_sample"
```

Comment:
curious why we do `sum(1)` instead of `1` or `1.0` here - doesn't really matter but I'm curious...